cli/src/commands/mod.rs

```diff
@@ -3812,24 +3813,44 @@ fn cmd_workspace_forget(
     args: &WorkspaceForgetArgs,
 ) -> Result<(), CommandError> {
     let mut workspace_command = command.workspace_helper(ui)?;
-
-    let workspace_id = if let Some(workspace_str) = &args.workspace {
-        WorkspaceId::new(workspace_str.to_string())
+    let mut wss = Vec::new();
+
+    let description = if args.workspaces.len() <= 1 {
+        // NOTE (aseipp): if there's only 1-or-0 arguments, shortcut. this is
+        // mostly so the oplog description can look good: it removes the need,
+        // in the case of more-than-1 argument, to handle pluralization of the
+        // nouns in the description
+        let ws = if args.workspaces.is_empty() {
+            workspace_command.workspace_id().to_owned()
+        } else {
+            WorkspaceId::new(args.workspaces[0].to_string())
```

Comment:
We need to actually validate this workspace name in this path, so this needs a slight refactoring.